.setup/bin/recreate_submissions.py

```diff
@@ -0,0 +1,294 @@
+    eg_vcs_host_type,eg_team_assignment,eg_max_team_size,eg_team_lock_date,
+    eg_use_ta_grading,eg_student_download,eg_student_view,
+    eg_student_view_after_grades,eg_student_submit,eg_submission_open_date,
+    eg_submission_due_date,eg_has_due_date,eg_late_days,
+    eg_allow_late_submission,eg_precision,eg_regrade_allowed,
+    eg_grade_inquiry_per_component_allowed,eg_grade_inquiry_due_date,
+    eg_thread_ids,eg_has_discussion,eg_limited_access_blind,eg_peer_blind,
+    eg_grade_inquiry_start_date,eg_hidden_files,eg_depends_on,
+    eg_depends_on_points,eg_has_release_date
+
+COURSE.electronic_gradeable_data
+- columns: g_id,user_id,team_id,g_version,
+    autograding_non_hidden_non_extra_credit,autograding_non_hidden_extra_credit,
+    autograding_hidden_non_extra_credit,autograding_hidden_extra_credit,
+    submission_time,autograding_complete
+- INSERT INTO electronic_gradeable_data (g_id,user_id,g_version,
+    submission_time,autograding_complete) VALUES (?,?,?,?,false)
+
+COURSE.electronic_gradeable_version
+- columns: g_id,user_id,team_id,active_version,anonymous_leaderboard
+- UPDATE electronic_gradeable_version SET active_version=?
+    WHERE g_id=? AND user_id=?
+'''
+
+import argparse
+import getpass
+import json
+import os
+import shutil
+import sqlalchemy as sql
+import sys
+import typing
+
+from submitty_utils import dateutils as submitty_dateutils
+
+# load configuration files
+with open('/usr/local/submitty/config/submitty.json','r') as conf:
+    SUBMITTY_CONFIG_JSON = json.load(conf)
+with open('/usr/local/submitty/config/database.json','r') as conf:
+    SUBMITTY_DBCONFIG_JSON = json.load(conf)
+
+ARG_INSTALL_DIR = SUBMITTY_CONFIG_JSON['submitty_install_dir']
+ARG_DATA_DIR = SUBMITTY_CONFIG_JSON['submitty_data_dir']
+#ARG_DB_HOST = 'localhost'
+ARG_DB_HOST = SUBMITTY_DBCONFIG_JSON['database_host']
+ARG_DB_PORT = SUBMITTY_DBCONFIG_JSON['database_port']
+ARG_DB_USER = SUBMITTY_DBCONFIG_JSON['database_user']
+ARG_DB_PASS = SUBMITTY_DBCONFIG_JSON['database_password']
+
+# variables specifying gradeable to submit to
+ARG_SEMESTER = ''
+ARG_COURSE = ''
+ARG_GRADEABLE = ''
+
+# place with files to submit
+ARG_SUBMISSIONS = ''
+
+# only use active versions?
+ARG_ACTIVE_ONLY = False
+
+# display extra output
+ARG_VERBOSE = False
+
+# some file names to look for
+UASF_NAME = 'user_assignment_settings.json'
+UAAF_NAME = '.user_assignment_access.json'
+UST_NAME = '.submit.timestamp'
+
+def submit(semester:str, course:str, gradeable:str, user:str, data: str,
+        func_add_submission: typing.Callable[[str,str,int,str],None],
+        func_update_active: typing.Callable[[str,str,int],None]):
+    '''
+    Create a new submission as a student
+    semester,course,gradeable = specify gradeable to make submission to
+    user = user to submit as
+    data = path to directory containing the files for submission
+    '''
+    gradeable_path = os.path.join(ARG_DATA_DIR,'courses',semester,course,
+        'submissions',gradeable)
```

Comment:
It would be better to use [pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os.path` throughout this script.

Reply:
What makes pathlib better?